src/main/java/ca/corefacility/bioinformatics/irida/ria/web/announcements/AnnouncementAjaxController.java

```diff
@@ -39,6 +43,48 @@ public AnnouncementAjaxController(UIAnnouncementsService UIAnnouncementsService)
 		return UIAnnouncementsService.getAnnouncementsAdmin(tableRequest);
 	}
 
+	/**
+	 * Handle request for getting a list of read announcements for a user.
+	 *
+	 * @param principal the currently logged in user
+	 * @return a {@link List} of unread {@link AnnouncementUserJoin}s for a user.
+	 */
+	@RequestMapping(value = "/user/read")
+	@PreAuthorize("hasRole('ROLE_USER')")
+	@ResponseBody
```

Comment:
Because this is a `RestController` you don't actually need to add @responsebody it does not hurt, but not required.  Just an FYI.

Reply:
Ok, thanks.